examples/trl/ddpo.py

```diff
@@ -0,0 +1,248 @@
+    scorer = AestheticScorer(
+        model_id=hub_model_id,
+        model_filename=model_filename,
+        dtype=torch.float32,
+    )
+    if use_habana:
+        scorer = scorer.to("hpu")
+    else:
+        scorer = scorer.to("cpu")
+
+    def _fn(images, prompts, metadata):
+        images = (images * 255).round().clamp(0, 255).to(torch.uint8)
+        scores = scorer(images)
+        return scores, {}
+
+    return _fn
+
+
+# list of example prompts to feed stable diffusion
+animals = [
+    "cat",
+    "dog",
+    "horse",
+    "monkey",
+    "rabbit",
+    "zebra",
+    "spider",
+    "bird",
+    "sheep",
+    "deer",
+    "cow",
+    "goat",
+    "lion",
+    "frog",
+    "chicken",
+    "duck",
+    "goose",
+    "bee",
+    "pig",
+    "turkey",
+    "fly",
+    "llama",
+    "camel",
+    "bat",
+    "gorilla",
+    "hedgehog",
+    "kangaroo",
+]
+
+
+def prompt_fn():
+    return np.random.choice(animals), {}
+
+
+def image_outputs_logger(image_data, global_step, accelerate_logger):
+    # For the sake of this example, we will only log the last batch of images
+    # and associated data
+    result = {}
+    images, prompts, _, rewards, _ = image_data[-1]
+
+    for i, image in enumerate(images):
+        prompt = prompts[i]
+        reward = rewards[i].item()
+        result[f"{prompt:.25} | {reward:.2f}"] = image.unsqueeze(0).float()
+
+    accelerate_logger.log_images(
+        result,
+        step=global_step,
+    )
+
+
+if __name__ == "__main__":
+    parser = HfArgumentParser((ScriptArguments, DDPOConfig))
+    args, ddpo_config = parser.parse_args_into_dataclasses()
+    ddpo_config.mixed_precision = "bf16" if args.bf16 else "no"
+    ddpo_config.project_kwargs = {
+        "logging_dir": "./logs",
+        "automatic_checkpoint_naming": True,
+        "total_limit": 5,
+        "project_dir": "./save",
+    }
+
+    # 1. initialize Gaudi config:
+    gaudi_config = GaudiConfig.from_pretrained(args.gaudi_config_name) if args.use_habana else None
+    gaudi_config.use_torch_autocast = gaudi_config.use_torch_autocast if args.bf16 else False
+
+    pipeline = GaudiDefaultDDPOStableDiffusionPipeline(
+        args.pretrained_model,
+        pretrained_model_revision=args.pretrained_revision,
+        use_lora=args.use_lora,
+        use_habana=args.use_habana,
+        use_hpu_graphs=args.use_hpu_graphs,
+        gaudi_config=gaudi_config,
+    )
+
+    trainer = GaudiDDPOTrainer(
+        ddpo_config,
+        aesthetic_scorer(
+            args.hf_hub_aesthetic_model_id,
+            args.hf_hub_aesthetic_model_filename,
+            args.use_habana,
+        ),
+        prompt_fn,
+        pipeline,
+        image_samples_hook=image_outputs_logger,
+        gaudi_config=gaudi_config,
+        use_habana=args.use_habana,
+        use_hpu_graphs=args.use_hpu_graphs,
+    )
+
+    trainer.train()
+
+    if args.push_to_hub:
+        trainer.push_to_hub(args.hf_hub_model_id, token=args.hf_user_access_token)
```

Comment:
`hf_user_access_token` is not defined in the args here, it is in the original example: https://github.com/huggingface/trl/blob/v0.7.8/examples/scripts/ddpo.py#L33

Reply:
@regisss I added the access token